docs/specification.md

```diff
@@ -0,0 +1,1280 @@
+  , address        :operator
+  , operatorTokens :tokens
+  )
+
+isOperatorResponse =
+  ( operatorParam :operator
+  , bool          :isOperator
+  )
+
+isOperatorParam =
+  ( operatorParam               :operator
+  , contract isOperatorResponse :callback
+  )
+
+isOperator = isOperatorParam
+```
+
+Parameter (in Michelson):
+```
+(pair %isOperator
+  (pair %operator
+    (address %owner)
+    (pair
+      (address %operator)
+      (or %tokens
+        (unit %allTokens)
+        (set %someTokens nat)
+      )
+  ))
+  (contract %callback
+    (pair
+      (pair %operator
+        (address %owner)
+        (pair
+          (address %operator)
+          (or %tokens
+            (unit %allTokens)
+            (set %someTokens nat)
+          )
+      ))
+      (bool %isOperator)
+    )
+  )
+)
+```
+
+- Inspect if an address is an operator for the specified owner and    token types.
+
+- If the adddress is not an operator for at least one
+  requested token type, then the result is `false`.
+
+- It's possible to make this query for some specific tokens, or to
+  all tokens.
+
+**Managed Ledger Functions (TBD)**
+----------------------------
+
+TODO: Maybe it's better to omit these mentions since all that left from
+ManagedLedger is their semantics?
+
+Functions for the TZUSDC token implementation found in Tezos [*ManagedLedger*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-7/ManagedLedger.md)
+
+**pause**
+
+Types
+```
+pause = (address :sender)
+```
+
+Parameter (in Michelson)
+```
+(pair %pause
+  (address %sender)
+  unit
+)
+```
+
+- Pauses transferring, burning, minting and allowance operations so
+  that they cannot be performed. All other operations remain
+  unaffected.
+
+- Cannot be called multiple times.
+
+- Sender must be a pauser.
+
+**unpause**
+
+Types
+```
+unpause = (address :sender)
+```
+
+Parameter (in Michelson)
+```
+(pair %unpause
+  (address %sender)
+  unit
+)
+```
+
+- Unpauses the contract so that transferring, burning, minting and
+  allowance operations can be performed by users with corresponding
+  roles.
+
+- Cannot be called multiple times.
+
+- Sender must be a pauser.
+
+**getPaused**
+
+Types
+```
+tokenId = nat
+
+getPausedResponse =
+  ( tokenId :tokenId
+  , nat     :paused
+  )
+
+getPausedParam =
+  ( list tokenId                      :tokenIds
+  , contract (list getPausedResponse) :callback
+  )
+
+getPaused = getPausedParam
+```
+
+Parameter (in Michelson)
+```
+(pair %getTotalPaused
+  (list %tokenIds nat)
+  (contract %callback
+    (list
+      (pair %getPausedResponse
+        (nat %tokenId)
+        (bool %paused)
+      )
+    )
+  )
+)
+```
+
+- Returns token pause status.
+
+- *Note: It's not a part of ManagedLedger interface*.
+
+**Custom token functions**
+==========================
+
+**Issuing and destroyng tokens**
+--------------------------
+
+**configureMinter**
+
+Types
+```
+tokenId = nat
+
+configureMinterParam =
+  ( address :sender
```

Comment:
Why do you need `sender` field here? Are you going to check that the transaction sender is equal to the `sender` in the parameter? In this case, you can just use `SENDER` instruction to get the sender value directly.

Or do you want an _operator_ to be able to call this function? I wonder if people would wish to offload such a sensitive action to operators.

Reply:
Actually I don't think that operators should be allowed call anything but `transfer`. I pass sender address here to verify if the given address is `masterMinter` and has `minter` address inside its minter list (or rather set?)

Reply:
UPD: to check if sender address is master minter, I have already answered earlier about why I explicitly pass this in documentation even so when I'm going to use `SENDER` in actual implementation